pkg/processor/runnable_processor.go

```diff
@@ -59,19 +62,41 @@ func (p *RunnableProcessor) Process(ctx context.Context, records []opencdc.Recor
 		p.inInsp.Send(ctx, record.FromOpenCDC(inRec))
 	}
 
-	outRecs := p.proc.Process(ctx, records)
-	for _, outRec := range outRecs {
-		singleRec, ok := outRec.(sdk.SingleRecord)
+	out := make([]sdk.ProcessedRecord, 0, len(records))
+	for _, rec := range records {
+		keep, err := p.evaluateCondition(rec)
+		if err != nil {
+			return append(out, sdk.ErrorRecord{Error: cerrors.Errorf("failed evaluating condition: %w", err)})
+		}
+		if !keep {
+			out = append(out, sdk.FilterRecord{})
```

Comment:
wouldn't this mean that we're filtering out the record when the condition is `false`? from what I understand we should append the original record here